support/trace/tracer.go

```diff
@@ -0,0 +1,79 @@
+package trace
+
+// CarrierFormat defines the format of the propogation that needs to injected or extracted
+type CarrierFormat byte
+
+const (
+	// Binary represents trace context as opaque binary data.
+	Binary CarrierFormat = iota
+	// TextMap represents trace context as key:value string pairs.
+	//
+	// Unlike HTTPHeaders, the TextMap format does not restrict the key or
+	// value character sets in any way.
+	TextMap
+	// HTTPHeaders represents trace context as HTTP header string pairs.
+	//
+	// Unlike TextMap, the HTTPHeaders format requires that the keys and values
+	// be valid as HTTP headers as-is (i.e., character casing may be unstable
+	// and special characters are disallowed in keys, values should be
+	// URL-escaped, etc).
+	HTTPHeaders
+)
+
+// FlogoTracer interface to configure individual tracers
+type FlogoTracer interface {
+	// Name() returns the name of the registered tracer
+	Name() string
+
+	// configure() sets up the tracer by gathering required configuration from the trace config
+	// environment variable. The tracer must also register as an event listenerbefore tracing can
+	// begin.
+	configure() error
+
+	// Inject() takes the `flowId` and `taskInstanceId` and injects the current span context for
+	// propagation within `carrier`. The actual type of `carrier` depends on the value of `format`.
+	//
+	// FlogoTracer defines a common set of `format` values, and each has an expected `carrier` type.
+	//
+	// Example usage:
+	//
+	// tracer := trace.GetTracer()
+	// ti := ctx.(*instance.TaskInst)
+	// err = tracer.Inject(ctx.ActivityHost().ID(), ti.InstanceId(), trace.HttpHeaders, req)
+	Inject(flowID string, taskInstanceID string, format CarrierFormat, carrier interface{}) error
```

Comment:
This api should be part of an Action specific Tracer, for example FlowTracer under the flow action.  Since not all actions have these properties.

Reply:
@fm-tibco : Can we just take pass the activity.Context here?